web_fragments/fragment.py

```diff
@@ -0,0 +1,271 @@
+"""
+Python representation of a web fragment.
+"""
+
+from collections import namedtuple
+
+FragmentResource = namedtuple("FragmentResource", "kind, data, mimetype, placement")  # pylint: disable=C0103
+
+JS_API_VERSION = 1
+
+
+class Fragment(object):
+    """
+    A fragment of a web page to be included on another page.
+
+    A fragment consists of HTML for the body of the page, and a series of
+    resources needed by the body. Resources are specified with a MIME type
+    (such as "application/javascript" or "text/css") that determines how they
+    are inserted into the page.  The resource is provided either as literal
+    text, or as a URL.  Text will be included on the page, wrapped
+    appropriately for the MIME type.  URLs will be used as-is on the page.
+
+    Resources are only inserted into the page once, even if many Fragments
+    in the page ask for them.  Determining duplicates is done by simple text
+    matching.
```

Comment:
@ormsbee @bjacobel +1 from me too. I won't do anything in this PR, but I strongly agree that we need to come up with a solid client-side story.